Software/src/main/java/org/foi/emp/carmanagement/configs/JwtAuthenticationEntryPoint.java

```diff
@@ -0,0 +1,18 @@
+package org.foi.emp.carmanagement.configs;
+
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.stereotype.Component;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.Serializable;
+
+@Component
+public class JwtAuthenticationEntryPoint implements AuthenticationEntryPoint, Serializable {
+
+    @Override
+    public void commence(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, org.springframework.security.core.AuthenticationException e) throws IOException {
```

Comment:
Could those variables be final?
`public void commence(final HttpServletRequest httpServletRequest, final HttpServletResponse httpServletResponse, final AuthenticationException authenticationException)`

if you can see the line above, i created an import. If it's going to work try to make custom import at the top so we can have cleaner code in function.

`import org.springframework.security.core.AuthenticationException `